astropy/wcs/tests/test_utils.py

```diff
@@ -353,6 +368,66 @@ def test_wcs_to_celestial_frame():
     assert isinstance(frame, Galactic)
 
 
+def test_wcs_to_body_frame():
+    unknown_wcs = WCS(naxis=2)
```

Comment:
Like for the other direction, I would move the tests for failures to the very end of the test function. I think it is more helpful if a test function first shows how things are supposed to work, and then checks that you get an appropriate error if you do something incorrectly.